### PR TITLE
SDA-4587 : add details to errors

### PR DESCRIPTION
**Pull request by ciaranRoche**
Part of : https://issues.redhat.com/browse/SDA-4587

This change adds details object to errors

### CODE REVIEW

File: pkg/generators/golang/errors_generator.go [hunk 1]
```diff
@@ -427,6 +454,9 @@ func (g *ErrorsGenerator) generateCommonErrors() error {
 				case "operation_id":
 					object.operationID = iterator.ReadString()
 					object.bitmap_ |= 16
+				case "details":
+					object.details = iterator.ReadAny()
```
Comment (ciaranRoche) [inline pkg/generators/golang/errors_generator.go:458]: @jhernand I am not sure if this is the correct function to call on an `interface{}` seems the best fit, as they dont have an explicit `ReadInterface()` func

Comment (jhernand) [reply pkg/generators/golang/errors_generator.go:458]: I think it is the right function. But that function returns an "Any" type. You will have to call "GetInterface" on that in order to convert it to the "interface{}" type used by "object.details".

**Review by jhernand — COMMENTED** [review]
I think the approach is correct, only needs a change to convert "Any" to "interface{}".

In addition I would like to include the new "details" field in these tests:

https://github.com/openshift-online/ocm-api-metamodel/blob/master/tests/go/builders_test.go#L207-L221
https://github.com/openshift-online/ocm-api-metamodel/blob/master/tests/go/marshal_test.go#L174-L194
https://github.com/openshift-online/ocm-api-metamodel/blob/master/tests/go/unmarshal_test.go#L466-L483

**Comment by ciaranRoche** [comment]
@jhernand tests added, and `getInterface{}` added. PTAL

**Review by jhernand — APPROVED** [review]
Looks good, thanks @ciaranRoche.

**Pull request merged**